File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/ContactColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContactRow } from "./ContactRow";

export interface ContactColumns {
    ContactId: Column<ContactRow>;
    Name: Column<ContactRow>;
    TaxCategoryName: Column<ContactRow>;
    TaxNumber: Column<ContactRow>;
}

export class ContactColumns extends ColumnsBase<ContactRow> {
    static readonly columnsKey = 'CrmDB.Contact';
    static readonly Fields = fieldsProxy<ContactColumns>();
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/Contact/ContactColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.Contact")]
[BasedOnRow(typeof(ContactRow), CheckNames = true)]
public class ContactColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ContactId { get; set; }
    [EditLink]
    public string Name { get; set; }

    public string TaxCategoryName { get; set; }

    public string TaxNumber { get; set; }
}